src/invidious/helpers/youtube_api.cr

```diff
@@ -98,13 +177,33 @@ end
 # The requested data is an endpoint (URL without the domain part)
 # and the data as a Hash object.
 #
-def _youtube_api_post_json(endpoint, data)
+def _youtube_api_post_json(endpoint, data, client_type = ApiClients::Web,
+                           proxy_region : String | Nil = nil) : Hash(String, JSON::Any)
+  client_key = HARDCODED_CLIENTS[client_type]["key"]
+  default_client_key = HARDCODED_CLIENTS[ApiClients::Web]["key"]
```

Comment:
You don't need a fallback key, tho. Your function definition sets `client_type` to `ApiClients::Web`, so it will use that if the argument isn't passed, and if you pass something other than `ApiClients::Web` or `ApiClients::Android`, the compiler will yell at you ^^